test/Driver/pipe_round_robin.swift.gyb

```diff
@@ -0,0 +1,48 @@
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/foo)
+// RUN: %empty-directory(%t/stats-pre)
+// RUN: %empty-directory(%t/stats-post)
+//
+// This test is looking at behaviour of the driver's task queue, checking
+// to make sure that it drains the output streams of multiple subprocesses
+// evenly, rather than one subprocess all-at-once before the next
+// all-at-once. This isn't batch-mode specific but it emerges somewhat
+// vividly there when combined with lots of files that do
+// -debug-time-function-bodies.
+//
+// Here we do a non-batch-mode variant that has lots of functions in each
+// of 4 files.
+//
+// RUN: %gyb -D N=1 %s -o %t/foo/file1.swift
+// RUN: %gyb -D N=2 %s -o %t/foo/file2.swift
+// RUN: %gyb -D N=3 %s -o %t/foo/file3.swift
+// RUN: %gyb -D N=4 %s -o %t/foo/file4.swift
+//
+// We have two runs here, one with and one without
+// -debug-time-function-bodies.
+//
+// We are trying to test that the presence of that flag does not radically
+// alter the CPU utilization of a run, but we want the test to be
+// reasonably robust against variation in code and CPU structure of the
+// machine we run the test on.
+//
+// Therefore we calculate the ratio of the user-delta and wall-delta
+// between runs; if the runs have "about the same" CPU utilization,
+// wall-delta/user-delta should be "around one", which we give lots of
+// wiggle room to, and write as abs(...) < 2.0.
+//
+// The extreme misbehaviour cases we were seeing had abs(...) ~= 4.0
+// (i.e. about the parallel-job count, as the misbehaviour causes
+// serialization) so this should catch it.
+//
+// RUN: %swiftc_driver -j 4 -module-name foo -c -stats-output-dir %t/stats-pre %t/foo/*.swift
+// RUN: %swiftc_driver -j 4 -module-name foo -c -stats-output-dir %t/stats-post -Xfrontend -debug-time-function-bodies %t/foo/*.swift
```

Comment:
These should be `%target-build-swift` so that they work for cross-compilation, and may as well use `-typecheck` instead of `-c` to do a bit less work.

Reply:
fixed.